programs/helium-entity-manager/src/instructions/initialize_shared_merkle_v0.rs

```diff
@@ -0,0 +1,84 @@
+use account_compression_cpi::{program::SplAccountCompression, Noop};
+use anchor_lang::prelude::*;
+use bubblegum_cpi::cpi::{accounts::CreateTree, create_tree};
+use bubblegum_cpi::program::Bubblegum;
+
+use crate::{shared_merkle_seeds, state::SharedMerkleV0};
+
+#[derive(AnchorSerialize, AnchorDeserialize, Clone, Default)]
+pub struct InitializeSharedMerkleArgsV0 {
+  pub proof_size: u8,
+}
+
+#[derive(Accounts)]
+#[instruction(args: InitializeSharedMerkleArgsV0)]
+pub struct InitializeSharedMerkleV0<'info> {
+  #[account(mut)]
+  pub payer: Signer<'info>,
```

Comment:
Let's scope this to the hprod signer. So not just anyone can mess with us when this goes live